Homework13/PageObjects/WebTablePage.cs

```diff
@@ -0,0 +1,126 @@
+        private MyWebElement _lastNameInput = new(By.XPath("//input[@id='lastName']"));
+        private MyWebElement _emailInput = new(By.XPath("//input[@id='userEmail']"));
+        private MyWebElement _ageInput = new(By.XPath("//input[@id='age']"));
+        private MyWebElement _salaryInput = new(By.XPath("//input[@id='salary']"));
+        private MyWebElement _departmentInput = new(By.XPath("//input[@id='department']"));
+        private MyWebElement _editButton = new(By.XPath("//span[@title='Edit']"));
+        private MyWebElement _deleteButton = new(By.XPath("//span[@id='delete-record-1']"));
+        private MyWebElement _closeButton = new(By.XPath("//span[contains(text(),'×')]"));
+
+        public void ClickAddNewEntryButton() => _addNewEntryButton.Click();
+
+        public void ClickCloseButton() => _closeButton.Click();
+
+        public void ClickSubmitButton() => _submitButton.Click();
+
+        public void ClickEditEntryButton() => _editButton.Click();
+
+        public void ClickDeleteEntryButton() => _deleteButton.Click();
+
+        public string GetFormClass() => _form.GetAttribute("class");
+
+        public string GetFirstName1Value() => _firstName1.Text;
+
+        public ReadOnlyCollection<IWebElement> GetRows()
+        {
+            var rows = _row.FindElements(By.XPath("//div[@role='rowgroup']"));
+
+            return rows;
+        }
+
+        public List<string> GetListOfNotEmptyRows(ReadOnlyCollection<IWebElement> rows)
+        {
+            List<string> rowsText = new List<string>();
+            for (int i = 0; i < rows.Count; i++)
+            {
+                var stringBuilder = new StringBuilder();
+                {
+                    IWebElement element = rows[i];
+                    string text = element.Text;
+                    stringBuilder.Append($"{text} ");
+                }
+                var finalString = stringBuilder.ToString();
+                if (finalString != "        ")
+                {
+                    rowsText.Add(finalString);
+                }
+            }
+
+            return rowsText;
+        }
+
+        public int GetCountOfFilteredRows(List<string> rows, string filterValue) => rows.Count(x => x.Contains(filterValue));
+
+        public int GetCountOfRows(List<string> rows) => rows.Count;
+
+        public string GetTextOfCollectionElements(ReadOnlyCollection<IWebElement> selectedItems)
+        {
+            var stringBuilder = new StringBuilder();
+            for (int i = 0; i < selectedItems.Count; i++)
+            {
+                IWebElement element = selectedItems[i];
+                string text = element.Text;
+                stringBuilder.Append($"{text} ");
+            }
+            var finalString = stringBuilder.ToString();
+
+            return finalString;
+        }
+
+        public void FillInSearchInput(string value) => _searchInput.SendKeys(value);
```

Comment:
Input is not valid for naming (TextBox or something else)

Reply:
updated